providers/common/include/internal/providercommon.h

```diff
@@ -8,7 +8,33 @@
  */
 
 #include <openssl/provider.h>
+#include <openssl/engine.h>
 
 const OSSL_PROVIDER *FIPS_get_provider(OPENSSL_CTX *ctx);
 
 const char *ossl_prov_util_nid_to_name(int nid);
+
+typedef struct {
+    /*
+     * References to the underlying digest implementation.  tmpmd caches
+     * the md, always.  alloc_md only holds a reference to an explicitly
+     * fetched digest.
+     * tmpmd is cleared after a CMAC_Init call.
+     */
+    const EVP_MD *tmpmd;         /* HMAC digest */
+    EVP_MD *alloc_md;            /* fetched digest */
+
+    /*
+     * Conditions for legacy EVP_MD uses.
+     * tmpengine is cleared after a CMAC_Init call.
+     */
+    ENGINE *tmpengine;           /* HMAC digest engine */
```

Comment:
HMAC?

Reply:
I didn't rework the comments -- yet.

Reply:
You missed the `CMAC_Init` in the comment two lines up :)